x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/AutodetectBuilder.java

```diff
@@ -251,8 +251,8 @@ public void build(int maxAnomalyRecords) throws IOException {
         return command;
     }
 
-    static String maxAnomalyRecordsArg(int maxAnomalyRecords) {
-        return "--maxAnomalyRecords=" + maxAnomalyRecords;
+    static String maxAnomalyRecordsArg(Settings settings) {
```

Comment:
Cool! Only thing is we now could make this method non-static and not have to pass in the settings. There are possibly more methods in there that could be non-static. Not necessary to do this in this PR though, only if you feel like it :-)